src/lib.rs

```diff
@@ -1,28 +1,123 @@
 use std::fmt::Write;
 
-#[derive(PartialEq, Eq)]
-pub struct Roff {
+/// Title line for a manpage.
+pub struct Title {
     title: String,
-    section: i8,
+    section: ManSection,
+    date: Option<String>,
+    source: Option<String>,
+    manual: Option<String>,
+}
+
+impl Title {
+    pub fn new(title: &str, section: ManSection) -> Self {
+        Title {
+            title: title.into(),
+            section,
+            date: None,
+            source: None,
+            manual: None,
+        }
+    }
+}
+
+impl Troffable for Title {
+    fn render(&self) -> String {
+        let manual = self.manual.as_deref().unwrap_or_default();
+        let date = self.date.as_deref().unwrap_or_default();
+        let source = self.source.as_deref().unwrap_or_default();
+
+        format!(
+            r#".TH "{}" "{}" "{}" "{}" "{}""#,
+            self.title.to_uppercase(),
+            self.section.value(),
+            date,
+            source,
+            manual
+        )
+    }
+}
+
+/// Manpage sections.
+///
+/// The most common is [`ManSection::Executable`], and is the recommended default.
+#[derive(Clone, Copy)]
+pub enum ManSection {
+    /// Executable programs or shell commands
+    Executable,
+    /// System calls (functions provided by the kernel)
+    SystemCalls,
+    /// Library calls (functions within program libraries)
+    LibraryCalls,
+    /// Special files (usually found in /dev)
+    SpecialFiles,
+    /// File formats and conventions, e.g. /etc/passwd
+    FileFormats,
+    /// Games
+    Games,
+    /// Miscellaneous (including macro packages and conventions), e.g. man(7), groff(7)
+    Miscellaneous,
+    /// System administration commands (usually only for root)
+    SystemAdministrationCommands,
+    /// Kernel routines [Non standard]
+    KernelRoutines,
+}
+
```

Comment:
Unless I'm missing something, this seems like man policy when this crate is focused on the underlying markup.

Reply:
> Your two comments about whether things should go here or in a more manpage specific crate was one of the reasons I ended up trying to do everything in `clap_man`, the inclusion of a title line means we're already targeting manpages and not "pure" `roff`. It is a _massive_ type-setting system ([here](https://man.archlinux.org/man/groff.7.en) is short reference of GNU variant of the format, you can see we're only really touching a tiny, tiny part of it), and has historically been used for lots of different things from books, articles and more.
> 
> Nowadays however I thing nobody uses it outside of manpages anymore, and while the `man` crate contains useful tools to build manpages for CLI tools this crate already targets a subset of `roff` specific to manpages, at least in my opinion. sweat_smile `roffman` that you mentioned in the PR in `clap` has [sections](https://github.com/vv9k/roffman/blob/master/src/lib.rs#L323) in it, though the name implies that it is specific for manpages, but not a huge fan of its API.
> 
> Sorry, this is a lot of words for saying that in my head `roff-rs` is already a manpage specific crate slightly_smiling_face It's again one of the reasons I've attempted a few different approaches to this, I haven't found a good solution for it yet besides creating my own crate for this.

My thought is to prefer to keep this crate man agnostic unless its getting in the way of getting `clap_man` done.

It looks like we could have the Section enum and the title formatting external to this crate and lose nothing *but* I'm not as knee deep in both crates as you are, so I defer to your judgement.

Reply:
Due to the current architecture this crate cannot be used to create anything besides manpages, so it is in a sense already specific to manpages and not agnostic. I'm starting to lean more towards having a more intermediate crate specifically for creating manpages, I have a prototype laying around that I could revive. Thoughts?